### PR TITLE
:book: Add automatic redirects for the development docs book to Netlify config

### PR DESCRIPTION
**Pull request by detiber**
**What this PR does / why we need it**:

Adds automated redirects for the master/development version of the Cluster API book

/hold
Depends on: https://github.com/kubernetes/k8s.io/pull/486

/assign @ncdc 
/kind documentation

### CODE REVIEW

File: netlify.toml [hunk 1]
```diff
@@ -5,20 +5,20 @@
...
+    from = "http://dev.cluster-api.sigs.k8s.io/*"
+    to = "https://dev.cluster-api.sigs.k8s.io/:splat"
     status = 301
     force = true
 
 [[redirects]]
-    from = "http://kubernetes-sigs-cluster-api.netlify.com/*"
-    to = "http://cluster-api.sigs.k8s.io/:splat"
+    from = "http://master--kubernetes-sigs-cluster-api.netlify.com/*"
+    to = "http://dev.cluster-api.sigs.k8s.io/:splat"
```
Comment (ncdc) [inline netlify.toml]: Would it save a redirect to make this https?

Comment (detiber) [reply netlify.toml]: Possibly, but I do know this is similar to the configuration we are using for the `release-0.2` branch and it is known to work :)

**Comment by ncdc** [comment]
My only remaining question is just confirming that we don't have anything that cares about `kubernetes-sigs-cluster-api.netlify.com`, since it's being dropped from this config?

**Comment by detiber** [comment]
> My only remaining question is just confirming that we don't have anything that cares about kubernetes-sigs-cluster-api.netlify.com, since it's being dropped from this config?

It's configured in the `netlify.toml` for the `release-0.2` branch, which owns that URL (as it is configured as the default branch in Netlify).

**Comment by ncdc** [comment]
Ok, so it won't be an issue that we're removing it from master, right?

**Comment by detiber** [comment]
> Ok, so it won't be an issue that we're removing it from master, right?

Correct.

**Comment by detiber** [comment]
@vincepri The DNS config has not been merged yet...

**Comment by ncdc** [comment]
I think the DNS change is in now?

**Comment by detiber** [comment]
It is, however I had forgotten that Netlify does not like the use of CNAME records where the dns subdomain used does not match the Netlify URL prefix for the branch. I've created https://github.com/kubernetes/k8s.io/pull/488 to update the DNS records properly and #1848 to re-add the proper redirects.

**Pull request merged**